### PR TITLE
Add FileTransferRequestEvent

### PR DESCRIPTION
**Pull request by avatus**
Supports: https://github.com/gravitational/teleport/pull/23546

This PR only adds a single event, but the proto changes were quite substantial so I'm breaking this chunk out to clean up the next, larger PRs

### CODE REVIEW

File: api/proto/teleport/legacy/types/events/events.proto [hunk 1]
```diff
@@ -601,6 +601,41 @@ message SessionConnect {
...
+
+  // SessionMetadata is a common event session metadata
+  SessionMetadata Session = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // RequestID is the ID for the FileTransferRequest
+  string RequestID = 3 [(gogoproto.jsontag) = "requestId"];
```
Comment (smallinsky) [inline api/proto/teleport/legacy/types/events/events.proto:621]:
```suggestion
  string RequestID = 3 [(gogoproto.jsontag) = "requestID"];
```

File: api/proto/teleport/legacy/types/events/events.proto [hunk 2]
```diff
@@ -601,6 +601,41 @@ message SessionConnect {
...
+  repeated string Approvers = 4 [(gogoproto.jsontag) = "approvers"];
+
+  // Requester is the Teleport user who requested the file transfer
+  string Requester = 5 [(gogoproto.jsontag) = "requester"];
+
+  // Location is the location of the file to be downloaded, or the directory of the upload
+  string Location = 6 [(gogoproto.jsontag) = "location"];
+
+  // Direction is "upload" or "download"
+  string Direction = 7 [(gogoproto.jsontag) = "direction"];
```
Comment (smallinsky) [inline api/proto/teleport/legacy/types/events/events.proto:633]: switch to enum type ?

File: api/proto/teleport/legacy/types/events/events.proto [hunk 3]
```diff
@@ -601,6 +601,41 @@ message SessionConnect {
...
+message FileTransferRequestEvent {
+  // Metadata is a common event metadata
+  Metadata Metadata = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // SessionMetadata is a common event session metadata
+  SessionMetadata Session = 2 [
```
Comment (smallinsky) [inline api/proto/teleport/legacy/types/events/events.proto:614]: Don't we want to include the user metadata?

Comment (avatus) [reply api/proto/teleport/legacy/types/events/events.proto:614]: We can but I don't see it being necessary at this point. The process doesn't really "care" who did it, as the approvers are stored in the request, as well as broadcasted (in the future PR) to the shared session on each action. This is mostly just an event to notify an updated happened to the state of the event.

File: api/proto/teleport/legacy/types/events/events.proto (hunk 2 above)
Comment (avatus) [reply api/proto/teleport/legacy/types/events/events.proto:633]: Sweet, idk why i didn't think of this. great idea!

File: api/proto/teleport/legacy/types/events/events.proto [hunk 4]
```diff
@@ -601,6 +601,46 @@ message SessionConnect {
...
+  repeated string Approvers = 4 [(gogoproto.jsontag) = "approvers"];
+
+  // Requester is the Teleport user who requested the file transfer
+  string Requester = 5 [(gogoproto.jsontag) = "requester"];
+
+  // Location is the location of the file to be downloaded, or the directory of the upload
+  string Location = 6 [(gogoproto.jsontag) = "location"];
+
+  // Direction is "upload" or "download"
+  FileTransferDirection Direction = 7 [(gogoproto.jsontag) = "direction"];
```
Comment (smallinsky) [inline api/proto/teleport/legacy/types/events/events.proto:633]: nit: Could you updated the comment and describe the the context of direction

File: api/proto/teleport/legacy/types/events/events.proto [hunk 5]
```diff
@@ -601,6 +601,46 @@ message SessionConnect {
...
+
+  // Filename is the name of the file to be uploaded to the Location. Only present in uploads.
+  string Filename = 8 [(gogoproto.jsontag) = "filename"];
+}
+
+enum FileTransferDirection {
+  DOWNLOAD = 0;
+  UPLOAD = 1;
+}
```
Comment (smallinsky) [inline api/proto/teleport/legacy/types/events/events.proto:642]: nit: missing go docs.

**Review by smallinsky — APPROVED** [review]
I haven't caught up the full context of the [RFD](https://github.com/gravitational/teleport/pull/23546) but per this protobuf change LGTM.

**Review by atburke — APPROVED** [review]

**Pull request merged**